ast/src/parsed/asm.rs

```diff
@@ -92,63 +92,86 @@ pub struct Import {
     pub path: SymbolPath,
 }
 
+/// A symbol path is a sequence of strings separated by ``::`.
+/// It can contain the special word `super`, which goes up a level.
+/// If it does not start with `::`, it is relative.
 #[derive(Default, Debug, PartialEq, Eq, Clone, PartialOrd, Ord)]
 pub struct SymbolPath {
-    pub parts: VecDeque<Part>,
+    /// The parts between each `::`.
+    pub parts: Vec<Part>,
 }
 
-#[derive(Default, Debug, PartialEq, Eq, Clone, PartialOrd, Ord)]
-pub struct AbsoluteSymbolPath {
-    pub parts: VecDeque<String>,
+impl SymbolPath {
+    pub fn join<P: Into<Self>>(mut self, other: P) -> Self {
+        self.parts.extend(other.into().parts);
+        self
+    }
 }
 
-impl<S: Into<String>> From<S> for AbsoluteSymbolPath {
-    fn from(name: S) -> Self {
-        Self {
-            parts: [name.into()].into(),
-        }
-    }
+/// An absolute symbol path is a resolved SymbolPath,
+/// which means it has to start with `::` and it cannot contain
+/// the word `super`.
```

Comment:
Why does it have to start with `::`?

Reply:
To distinguish it syntactically from a relative path. That way we can mix absolute and relative paths and don't have to provide a boolean flag to indicate the way we interpret paths.